Generated struct named `gaussianParticle_RawParameters_0` mismatches its usage

Hi,

While run the code, I noticed a discrepancy between the definition and the actual usage of the struct `gaussianParticle_RawParameters_0`. This mismatch is causing a compilation error during the training of 3DGUT.

# Steps to Reproduce

```bash
python train.py --config-name apps/nerf_synthetic_3dgut.yaml path=data/nerf_synthetic/lego out_dir=runs experiment_name=lego_3dgut
```
# Expected Behavior

In compiling native code stage, build extension module `lib3dgut_cc` should be successful.

# Actual Behavior

```plain
$ python train.py --config-name apps/nerf_synthetic_3dgut.yaml path=data/nerf_synthetic/lego out_dir=runs experiment_name=lego_3dgut
[11:50:46] [INFO] Compiling native code..                                                                                                                                                                        logger.py:60
─────────────────────────────────────────────────────────────────────────────────────────────────────── Load Datasets ───────────────────────────────────────────────────────────────────────────────────────────────────────
Load Dataset (train) ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% :: 🕒 Elapsed 0:00:03
Load Dataset (val) ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% :: 🕒 Elapsed 0:00:03
───────────────────────────────────────────────────────────────────────────────────────────────────── Initialize Model ──────────────────────────────────────────────────────────────────────────────────────────────────────
Using <DELETED>/.cache/torch_extensions/py311_cu118 as PyTorch extensions root...
Creating extension directory <DELETED>/.cache/torch_extensions/py311_cu118/lib3dgut_cc...
Detected CUDA files, patching ldflags
Emitting ninja build file <DELETED>/.cache/torch_extensions/py311_cu118/lib3dgut_cc/build.ninja...
Building extension module lib3dgut_cc...
Allowing ninja to set a default number of workers... (overridable by setting the environment variable MAX_JOBS=N)
[1/5] /usr/bin/g++-11 -MMD -MF cudaBuffer.o.d -DTORCH_EXTENSION_NAME=lib3dgut_cc -DTORCH_API_INCLUDE_EXTENSION_H -DPYBIND11_COMPILER_TYPE=\"_gcc\" -DPYBIND11_STDLIB=\"_libstdcpp\" -DPYBIND11_BUILD_ABI=\"_cxxabi1011\" -I<DELETED>/3dgrut/threedgut_tracer/include -I<DELETED>/3dgrut/thirdparty/tiny-cuda-nn/include -I<DELETED>/3dgrut/thirdparty/tiny-cuda-nn/dependencies -I<DELETED>/3dgrut/thirdparty/tiny-cuda-nn/dependencies/fmt/include -I<DELETED>/.cache/torch_extensions/py311_cu118/lib3dgut_cc -isystem <DELETED>/.conda/envs/3dgrut/lib/python3.11/site-packages/torch/include -isystem <DELETED>/.conda/envs/3dgrut/lib/python3.11/site-packages/torch/include/torch/csrc/api/include -isystem <DELETED>/.conda/envs/3dgrut/lib/python3.11/site-packages/torch/include/TH -isystem <DELETED>/.conda/envs/3dgrut/lib/python3.11/site-packages/torch/include/THC -isystem /usr/local/cuda-11.8/include -isystem <DELETED>/.conda/envs/3dgrut/include/python3.11 -D_GLIBCXX_USE_CXX11_ABI=0 -fPIC -std=c++17 -DNVDR_TORCH -DTCNN_MIN_GPU_ARCH=70 -DPARTICLE_RADIANCE_NUM_COEFFS=16 -DGAUSSIAN_PARTICLE_KERNEL_DEGREE=2 -DGAUSSIAN_PARTICLE_MIN_KERNEL_DENSITY=0.0 -DGAUSSIAN_PARTICLE_MIN_ALPHA=0.00392156862745098 -DGAUSSIAN_PARTICLE_MAX_ALPHA=0.99 -DGAUSSIAN_MIN_TRANSMITTANCE_THRESHOLD=0.0001 -DGAUSSIAN_N_ROLLING_SHUTTER_ITERATIONS=5 -DGAUSSIAN_K_BUFFER_SIZE=0 -DGAUSSIAN_GLOBAL_Z_ORDER=true -DGAUSSIAN_UT_ALPHA=1.0 -DGAUSSIAN_UT_BETA=2.0 -DGAUSSIAN_UT_KAPPA=0.0 -DGAUSSIAN_UT_DELTA=1.7320508075688772 -DGAUSSIAN_UT_IN_IMAGE_MARGIN_FACTOR=0.1 -DGAUSSIAN_UT_REQUIRE_ALL_SIGMA_POINTS_VALID=false -DGAUSSIAN_RECT_BOUNDING=true -DGAUSSIAN_TIGHT_OPACITY_BOUNDING=true -DGAUSSIAN_TILE_BASED_CULLING=true -c <DELETED>/3dgrut/threedgut_tracer/src/cudaBuffer.cpp -o cudaBuffer.o 
[2/5] /usr/local/cuda-11.8/bin/nvcc  -ccbin /usr/bin/gcc-11 -DTORCH_EXTENSION_NAME=lib3dgut_cc -DTORCH_API_INCLUDE_EXTENSION_H -DPYBIND11_COMPILER_TYPE=\"_gcc\" -DPYBIND11_STDLIB=\"_libstdcpp\" -DPYBIND11_BUILD_ABI=\"_cxxabi1011\" -I<DELETED>/3dgrut/threedgut_tracer/include -I<DELETED>/3dgrut/thirdparty/tiny-cuda-nn/include -I<DELETED>/3dgrut/thirdparty/tiny-cuda-nn/dependencies -I<DELETED>/3dgrut/thirdparty/tiny-cuda-nn/dependencies/fmt/include -I<DELETED>/.cache/torch_extensions/py311_cu118/lib3dgut_cc -isystem <DELETED>/.conda/envs/3dgrut/lib/python3.11/site-packages/torch/include -isystem <DELETED>/.conda/envs/3dgrut/lib/python3.11/site-packages/torch/include/torch/csrc/api/include -isystem <DELETED>/.conda/envs/3dgrut/lib/python3.11/site-packages/torch/include/TH -isystem <DELETED>/.conda/envs/3dgrut/lib/python3.11/site-packages/torch/include/THC -isystem /usr/local/cuda-11.8/include -isystem <DELETED>/.conda/envs/3dgrut/include/python3.11 -D_GLIBCXX_USE_CXX11_ABI=0 -D__CUDA_NO_HALF_OPERATORS__ -D__CUDA_NO_HALF_CONVERSIONS__ -D__CUDA_NO_BFLOAT16_CONVERSIONS__ -D__CUDA_NO_HALF2_OPERATORS__ --expt-relaxed-constexpr -gencode=arch=compute_89,code=compute_89 -gencode=arch=compute_89,code=sm_89 --compiler-options '-fPIC' -DNVDR_TORCH -DTCNN_MIN_GPU_ARCH=70 -std=c++17 --extended-lambda --expt-relaxed-constexpr -Xcompiler=-Wno-float-conversion -Xcompiler=-fno-strict-aliasing -U__CUDA_NO_HALF_OPERATORS__ -U__CUDA_NO_HALF_CONVERSIONS__ -U__CUDA_NO_BFLOAT16_CONVERSIONS__ -U__CUDA_NO_HALF2_OPERATORS__ -use_fast_math -DPARTICLE_RADIANCE_NUM_COEFFS=16 -DGAUSSIAN_PARTICLE_KERNEL_DEGREE=2 -DGAUSSIAN_PARTICLE_MIN_KERNEL_DENSITY=0.0 -DGAUSSIAN_PARTICLE_MIN_ALPHA=0.00392156862745098 -DGAUSSIAN_PARTICLE_MAX_ALPHA=0.99 -DGAUSSIAN_MIN_TRANSMITTANCE_THRESHOLD=0.0001 -DGAUSSIAN_N_ROLLING_SHUTTER_ITERATIONS=5 -DGAUSSIAN_K_BUFFER_SIZE=0 -DGAUSSIAN_GLOBAL_Z_ORDER=true -DGAUSSIAN_UT_ALPHA=1.0 -DGAUSSIAN_UT_BETA=2.0 -DGAUSSIAN_UT_KAPPA=0.0 -DGAUSSIAN_UT_DELTA=1.7320508075688772 -DGAUSSIAN_UT_IN_IMAGE_MARGIN_FACTOR=0.1 -DGAUSSIAN_UT_REQUIRE_ALL_SIGMA_POINTS_VALID=false -DGAUSSIAN_RECT_BOUNDING=true -DGAUSSIAN_TIGHT_OPACITY_BOUNDING=true -DGAUSSIAN_TILE_BASED_CULLING=true -c <DELETED>/3dgrut/threedgut_tracer/src/gutRenderer.cu -o gutRenderer.cuda.o 
FAILED: gutRenderer.cuda.o 
/usr/local/cuda-11.8/bin/nvcc  -ccbin /usr/bin/gcc-11 -DTORCH_EXTENSION_NAME=lib3dgut_cc -DTORCH_API_INCLUDE_EXTENSION_H -DPYBIND11_COMPILER_TYPE=\"_gcc\" -DPYBIND11_STDLIB=\"_libstdcpp\" -DPYBIND11_BUILD_ABI=\"_cxxabi1011\" -I<DELETED>/3dgrut/threedgut_tracer/include -I<DELETED>/3dgrut/thirdparty/tiny-cuda-nn/include -I<DELETED>/3dgrut/thirdparty/tiny-cuda-nn/dependencies -I<DELETED>/3dgrut/thirdparty/tiny-cuda-nn/dependencies/fmt/include -I<DELETED>/.cache/torch_extensions/py311_cu118/lib3dgut_cc -isystem <DELETED>/.conda/envs/3dgrut/lib/python3.11/site-packages/torch/include -isystem <DELETED>/.conda/envs/3dgrut/lib/python3.11/site-packages/torch/include/torch/csrc/api/include -isystem <DELETED>/.conda/envs/3dgrut/lib/python3.11/site-packages/torch/include/TH -isystem <DELETED>/.conda/envs/3dgrut/lib/python3.11/site-packages/torch/include/THC -isystem /usr/local/cuda-11.8/include -isystem <DELETED>/.conda/envs/3dgrut/include/python3.11 -D_GLIBCXX_USE_CXX11_ABI=0 -D__CUDA_NO_HALF_OPERATORS__ -D__CUDA_NO_HALF_CONVERSIONS__ -D__CUDA_NO_BFLOAT16_CONVERSIONS__ -D__CUDA_NO_HALF2_OPERATORS__ --expt-relaxed-constexpr -gencode=arch=compute_89,code=compute_89 -gencode=arch=compute_89,code=sm_89 --compiler-options '-fPIC' -DNVDR_TORCH -DTCNN_MIN_GPU_ARCH=70 -std=c++17 --extended-lambda --expt-relaxed-constexpr -Xcompiler=-Wno-float-conversion -Xcompiler=-fno-strict-aliasing -U__CUDA_NO_HALF_OPERATORS__ -U__CUDA_NO_HALF_CONVERSIONS__ -U__CUDA_NO_BFLOAT16_CONVERSIONS__ -U__CUDA_NO_HALF2_OPERATORS__ -use_fast_math -DPARTICLE_RADIANCE_NUM_COEFFS=16 -DGAUSSIAN_PARTICLE_KERNEL_DEGREE=2 -DGAUSSIAN_PARTICLE_MIN_KERNEL_DENSITY=0.0 -DGAUSSIAN_PARTICLE_MIN_ALPHA=0.00392156862745098 -DGAUSSIAN_PARTICLE_MAX_ALPHA=0.99 -DGAUSSIAN_MIN_TRANSMITTANCE_THRESHOLD=0.0001 -DGAUSSIAN_N_ROLLING_SHUTTER_ITERATIONS=5 -DGAUSSIAN_K_BUFFER_SIZE=0 -DGAUSSIAN_GLOBAL_Z_ORDER=true -DGAUSSIAN_UT_ALPHA=1.0 -DGAUSSIAN_UT_BETA=2.0 -DGAUSSIAN_UT_KAPPA=0.0 -DGAUSSIAN_UT_DELTA=1.7320508075688772 -DGAUSSIAN_UT_IN_IMAGE_MARGIN_FACTOR=0.1 -DGAUSSIAN_UT_REQUIRE_ALL_SIGMA_POINTS_VALID=false -DGAUSSIAN_RECT_BOUNDING=true -DGAUSSIAN_TIGHT_OPACITY_BOUNDING=true -DGAUSSIAN_TILE_BASED_CULLING=true -c <DELETED>/3dgrut/threedgut_tracer/src/gutRenderer.cu -o gutRenderer.cuda.o 
<DELETED>/3dgrut/thirdparty/tiny-cuda-nn/dependencies/fmt/include/fmt/core.h(288): warning #1675-D: unrecognized GCC pragma

<DELETED>/3dgrut/threedgut_tracer/include/3dgut/kernels/cuda/renderers/gutProjector.cuh(305): warning #174-D: expression has no effect

<DELETED>/3dgrut/threedgut_tracer/include/3dgut/kernels/cuda/models/shRadiativeGaussianParticles.cuh(89): error: class "gaussianParticle_Parameters_0" has no member "density_1"
          detected during:
            instantiation of "const float &ShRadiativeGaussianVolumetricFeaturesParticles<TDensityParameters, TFeaturesParameters, Params, ExtParams, TDifferentiable>::opacity(const TDensityParameters &) const [with TDensityParameters=gaussianParticle_Parameters_0, TFeaturesParameters=shRadiativeParticle_Parameters_0, Params=model_InternalParams, ExtParams=model_ExternalParams, TDifferentiable=true]" 
<DELETED>/3dgrut/threedgut_tracer/include/3dgut/kernels/cuda/renderers/gutProjector.cuh(131): here
            instantiation of "__nv_bool GUTProjector<Particles, Params, UTParams>::unscentedParticleProjection(const tcnn::ivec2 &, const threedgut::TSensorModel &, const tcnn::vec3 &, const tcnn::mat4x3 &, const threedgut::TSensorState &, const Particles &, const Particles::DensityParameters &, tcnn::vec3 &, float &, tcnn::vec2 &, tcnn::vec3 &) [with Particles=model_Particles, Params=TGUTProjectorParams, UTParams=TGUTProjectionParams]" 
<DELETED>/3dgrut/threedgut_tracer/include/3dgut/kernels/cuda/renderers/gutProjector.cuh(249): here
            instantiation of "void GUTProjector<Particles, Params, UTParams>::eval(tcnn::uvec2, uint32_t, tcnn::ivec2, threedgut::TSensorModel, tcnn::vec3, tcnn::mat4x3, threedgut::TSensorState, uint32_t *, tcnn::vec2 *, tcnn::vec4 *, tcnn::vec2 *, float *, float *, threedgut::MemoryHandles) [with Particles=model_Particles, Params=TGUTProjectorParams, UTParams=TGUTProjectionParams]" 
<DELETED>/3dgrut/threedgut_tracer/include/3dgut/kernels/cuda/renderers/gutRenderer.cuh(36): here

<DELETED>/3dgrut/threedgut_tracer/include/3dgut/kernels/cuda/models/shRadiativeGaussianParticles.cuh(76): error: class "gaussianParticle_Parameters_0" has no member "position_1"
          detected during:
            instantiation of "const tcnn::vec3 &ShRadiativeGaussianVolumetricFeaturesParticles<TDensityParameters, TFeaturesParameters, Params, ExtParams, TDifferentiable>::position(const TDensityParameters &) const [with TDensityParameters=gaussianParticle_Parameters_0, TFeaturesParameters=shRadiativeParticle_Parameters_0, Params=model_InternalParams, ExtParams=model_ExternalParams, TDifferentiable=true]" 
<DELETED>/3dgrut/threedgut_tracer/include/3dgut/kernels/cuda/renderers/gutProjector.cuh(136): here
            instantiation of "__nv_bool GUTProjector<Particles, Params, UTParams>::unscentedParticleProjection(const tcnn::ivec2 &, const threedgut::TSensorModel &, const tcnn::vec3 &, const tcnn::mat4x3 &, const threedgut::TSensorState &, const Particles &, const Particles::DensityParameters &, tcnn::vec3 &, float &, tcnn::vec2 &, tcnn::vec3 &) [with Particles=model_Particles, Params=TGUTProjectorParams, UTParams=TGUTProjectionParams]" 
<DELETED>/3dgrut/threedgut_tracer/include/3dgut/kernels/cuda/renderers/gutProjector.cuh(249): here
            instantiation of "void GUTProjector<Particles, Params, UTParams>::eval(tcnn::uvec2, uint32_t, tcnn::ivec2, threedgut::TSensorModel, tcnn::vec3, tcnn::mat4x3, threedgut::TSensorState, uint32_t *, tcnn::vec2 *, tcnn::vec4 *, tcnn::vec2 *, float *, float *, threedgut::MemoryHandles) [with Particles=model_Particles, Params=TGUTProjectorParams, UTParams=TGUTProjectionParams]" 
<DELETED>/3dgrut/threedgut_tracer/include/3dgut/kernels/cuda/renderers/gutRenderer.cuh(36): here

<DELETED>/3dgrut/threedgut_tracer/include/3dgut/kernels/cuda/models/shRadiativeGaussianParticles.cuh(80): error: class "gaussianParticle_Parameters_0" has no member "scale_1"
          detected during:
            instantiation of "const tcnn::vec3 &ShRadiativeGaussianVolumetricFeaturesParticles<TDensityParameters, TFeaturesParameters, Params, ExtParams, TDifferentiable>::scale(const TDensityParameters &) const [with TDensityParameters=gaussianParticle_Parameters_0, TFeaturesParameters=shRadiativeParticle_Parameters_0, Params=model_InternalParams, ExtParams=model_ExternalParams, TDifferentiable=true]" 
<DELETED>/3dgrut/threedgut_tracer/include/3dgut/kernels/cuda/renderers/gutProjector.cuh(142): here
            instantiation of "__nv_bool GUTProjector<Particles, Params, UTParams>::unscentedParticleProjection(const tcnn::ivec2 &, const threedgut::TSensorModel &, const tcnn::vec3 &, const tcnn::mat4x3 &, const threedgut::TSensorState &, const Particles &, const Particles::DensityParameters &, tcnn::vec3 &, float &, tcnn::vec2 &, tcnn::vec3 &) [with Particles=model_Particles, Params=TGUTProjectorParams, UTParams=TGUTProjectionParams]" 
<DELETED>/3dgrut/threedgut_tracer/include/3dgut/kernels/cuda/renderers/gutProjector.cuh(249): here
            instantiation of "void GUTProjector<Particles, Params, UTParams>::eval(tcnn::uvec2, uint32_t, tcnn::ivec2, threedgut::TSensorModel, tcnn::vec3, tcnn::mat4x3, threedgut::TSensorState, uint32_t *, tcnn::vec2 *, tcnn::vec4 *, tcnn::vec2 *, float *, float *, threedgut::MemoryHandles) [with Particles=model_Particles, Params=TGUTProjectorParams, UTParams=TGUTProjectionParams]" 
<DELETED>/3dgrut/threedgut_tracer/include/3dgut/kernels/cuda/renderers/gutRenderer.cuh(36): here

<DELETED>/3dgrut/threedgut_tracer/include/3dgut/kernels/cuda/models/shRadiativeGaussianParticles.cuh(72): error: class "gaussianParticle_RawParameters_0" has no member "position_0"
          detected during:
            instantiation of "tcnn::vec3 ShRadiativeGaussianVolumetricFeaturesParticles<TDensityParameters, TFeaturesParameters, Params, ExtParams, TDifferentiable>::fetchPosition(uint32_t) const [with TDensityParameters=gaussianParticle_Parameters_0, TFeaturesParameters=shRadiativeParticle_Parameters_0, Params=model_InternalParams, ExtParams=model_ExternalParams, TDifferentiable=true]" 
<DELETED>/3dgrut/threedgut_tracer/include/3dgut/kernels/cuda/renderers/gutProjector.cuh(415): here
            instantiation of "void GUTProjector<Particles, Params, UTParams>::evalBackward(tcnn::uvec2, uint32_t, tcnn::ivec2, threedgut::TSensorModel, tcnn::vec3, tcnn::mat4x3, const uint32_t *, threedgut::MemoryHandles, const tcnn::vec2 *, const tcnn::vec4 *, const float *, const float *, const float *, threedgut::MemoryHandles) [with Particles=model_Particles, Params=TGUTProjectorParams, UTParams=TGUTProjectionParams]" 
<DELETED>/3dgrut/threedgut_tracer/include/3dgut/kernels/cuda/renderers/gutRenderer.cuh(179): here

<DELETED>
.
.
.
<DELETED>

42 errors detected in the compilation of "<DELETED>/3dgrut/threedgut_tracer/src/gutRenderer.cu".
[3/5] /usr/bin/g++-11 -MMD -MF bindings.o.d -DTORCH_EXTENSION_NAME=lib3dgut_cc -DTORCH_API_INCLUDE_EXTENSION_H -DPYBIND11_COMPILER_TYPE=\"_gcc\" -DPYBIND11_STDLIB=\"_libstdcpp\" -DPYBIND11_BUILD_ABI=\"_cxxabi1011\" -I<DELETED>/3dgrut/threedgut_tracer/include -I<DELETED>/3dgrut/thirdparty/tiny-cuda-nn/include -I<DELETED>/3dgrut/thirdparty/tiny-cuda-nn/dependencies -I<DELETED>/3dgrut/thirdparty/tiny-cuda-nn/dependencies/fmt/include -I<DELETED>/.cache/torch_extensions/py311_cu118/lib3dgut_cc -isystem <DELETED>/.conda/envs/3dgrut/lib/python3.11/site-packages/torch/include -isystem <DELETED>/.conda/envs/3dgrut/lib/python3.11/site-packages/torch/include/torch/csrc/api/include -isystem <DELETED>/.conda/envs/3dgrut/lib/python3.11/site-packages/torch/include/TH -isystem <DELETED>/.conda/envs/3dgrut/lib/python3.11/site-packages/torch/include/THC -isystem /usr/local/cuda-11.8/include -isystem <DELETED>/.conda/envs/3dgrut/include/python3.11 -D_GLIBCXX_USE_CXX11_ABI=0 -fPIC -std=c++17 -DNVDR_TORCH -DTCNN_MIN_GPU_ARCH=70 -DPARTICLE_RADIANCE_NUM_COEFFS=16 -DGAUSSIAN_PARTICLE_KERNEL_DEGREE=2 -DGAUSSIAN_PARTICLE_MIN_KERNEL_DENSITY=0.0 -DGAUSSIAN_PARTICLE_MIN_ALPHA=0.00392156862745098 -DGAUSSIAN_PARTICLE_MAX_ALPHA=0.99 -DGAUSSIAN_MIN_TRANSMITTANCE_THRESHOLD=0.0001 -DGAUSSIAN_N_ROLLING_SHUTTER_ITERATIONS=5 -DGAUSSIAN_K_BUFFER_SIZE=0 -DGAUSSIAN_GLOBAL_Z_ORDER=true -DGAUSSIAN_UT_ALPHA=1.0 -DGAUSSIAN_UT_BETA=2.0 -DGAUSSIAN_UT_KAPPA=0.0 -DGAUSSIAN_UT_DELTA=1.7320508075688772 -DGAUSSIAN_UT_IN_IMAGE_MARGIN_FACTOR=0.1 -DGAUSSIAN_UT_REQUIRE_ALL_SIGMA_POINTS_VALID=false -DGAUSSIAN_RECT_BOUNDING=true -DGAUSSIAN_TIGHT_OPACITY_BOUNDING=true -DGAUSSIAN_TILE_BASED_CULLING=true -c <DELETED>/3dgrut/threedgut_tracer/bindings.cpp -o bindings.o 
[4/5] /usr/local/cuda-11.8/bin/nvcc  -ccbin /usr/bin/gcc-11 -DTORCH_EXTENSION_NAME=lib3dgut_cc -DTORCH_API_INCLUDE_EXTENSION_H -DPYBIND11_COMPILER_TYPE=\"_gcc\" -DPYBIND11_STDLIB=\"_libstdcpp\" -DPYBIND11_BUILD_ABI=\"_cxxabi1011\" -I<DELETED>/3dgrut/threedgut_tracer/include -I<DELETED>/3dgrut/thirdparty/tiny-cuda-nn/include -I<DELETED>/3dgrut/thirdparty/tiny-cuda-nn/dependencies -I<DELETED>/3dgrut/thirdparty/tiny-cuda-nn/dependencies/fmt/include -I<DELETED>/.cache/torch_extensions/py311_cu118/lib3dgut_cc -isystem <DELETED>/.conda/envs/3dgrut/lib/python3.11/site-packages/torch/include -isystem <DELETED>/.conda/envs/3dgrut/lib/python3.11/site-packages/torch/include/torch/csrc/api/include -isystem <DELETED>/.conda/envs/3dgrut/lib/python3.11/site-packages/torch/include/TH -isystem <DELETED>/.conda/envs/3dgrut/lib/python3.11/site-packages/torch/include/THC -isystem /usr/local/cuda-11.8/include -isystem <DELETED>/.conda/envs/3dgrut/include/python3.11 -D_GLIBCXX_USE_CXX11_ABI=0 -D__CUDA_NO_HALF_OPERATORS__ -D__CUDA_NO_HALF_CONVERSIONS__ -D__CUDA_NO_BFLOAT16_CONVERSIONS__ -D__CUDA_NO_HALF2_OPERATORS__ --expt-relaxed-constexpr -gencode=arch=compute_89,code=compute_89 -gencode=arch=compute_89,code=sm_89 --compiler-options '-fPIC' -DNVDR_TORCH -DTCNN_MIN_GPU_ARCH=70 -std=c++17 --extended-lambda --expt-relaxed-constexpr -Xcompiler=-Wno-float-conversion -Xcompiler=-fno-strict-aliasing -U__CUDA_NO_HALF_OPERATORS__ -U__CUDA_NO_HALF_CONVERSIONS__ -U__CUDA_NO_BFLOAT16_CONVERSIONS__ -U__CUDA_NO_HALF2_OPERATORS__ -use_fast_math -DPARTICLE_RADIANCE_NUM_COEFFS=16 -DGAUSSIAN_PARTICLE_KERNEL_DEGREE=2 -DGAUSSIAN_PARTICLE_MIN_KERNEL_DENSITY=0.0 -DGAUSSIAN_PARTICLE_MIN_ALPHA=0.00392156862745098 -DGAUSSIAN_PARTICLE_MAX_ALPHA=0.99 -DGAUSSIAN_MIN_TRANSMITTANCE_THRESHOLD=0.0001 -DGAUSSIAN_N_ROLLING_SHUTTER_ITERATIONS=5 -DGAUSSIAN_K_BUFFER_SIZE=0 -DGAUSSIAN_GLOBAL_Z_ORDER=true -DGAUSSIAN_UT_ALPHA=1.0 -DGAUSSIAN_UT_BETA=2.0 -DGAUSSIAN_UT_KAPPA=0.0 -DGAUSSIAN_UT_DELTA=1.7320508075688772 -DGAUSSIAN_UT_IN_IMAGE_MARGIN_FACTOR=0.1 -DGAUSSIAN_UT_REQUIRE_ALL_SIGMA_POINTS_VALID=false -DGAUSSIAN_RECT_BOUNDING=true -DGAUSSIAN_TIGHT_OPACITY_BOUNDING=true -DGAUSSIAN_TILE_BASED_CULLING=true -c <DELETED>/3dgrut/threedgut_tracer/src/splatRaster.cu -o splatRaster.cuda.o 
ninja: build stopped: subcommand failed.
Error executing job with overrides: ['path=data/nerf_synthetic/lego', 'out_dir=runs', 'experiment_name=lego_3dgut']
Traceback (most recent call last):
  File "<DELETED>/3dgrut/threedgut_tracer/tracer.py", line 41, in load_3dgut_plugin
    from . import lib3dgut_cc as tdgut  # type: ignore
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ImportError: cannot import name 'lib3dgut_cc' from 'threedgut_tracer' (<DELETED>/3dgrut/threedgut_tracer/__init__.py)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<DELETED>/.conda/envs/3dgrut/lib/python3.11/site-packages/torch/utils/cpp_extension.py", line 2100, in _run_ninja_build
    subprocess.run(
  File "<DELETED>/.conda/envs/3dgrut/lib/python3.11/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['ninja', '-v']' returned non-zero exit status 1.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<DELETED>/3dgrut/train.py", line 39, in main
    trainer = Trainer3DGRUT(conf)
              ^^^^^^^^^^^^^^^^^^^
  File "<DELETED>/.conda/envs/3dgrut/lib/python3.11/contextlib.py", line 81, in inner
    return func(*args, **kwds)
           ^^^^^^^^^^^^^^^^^^^
  File "<DELETED>/3dgrut/threedgrut/trainer.py", line 128, in __init__
    self.init_model(conf, self.scene_extent)
  File "<DELETED>/3dgrut/threedgrut/trainer.py", line 178, in init_model
    self.model = MixtureOfGaussians(conf, scene_extent=scene_extent)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<DELETED>/3dgrut/threedgrut/model/model.py", line 154, in __init__
    self.renderer = threedgut_tracer.Tracer(conf)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<DELETED>/3dgrut/threedgut_tracer/tracer.py", line 289, in __init__
    load_3dgut_plugin(conf)
  File "<DELETED>/3dgrut/threedgut_tracer/tracer.py", line 45, in load_3dgut_plugin
    setup_3dgut(conf)
  File "<DELETED>/3dgrut/threedgut_tracer/setup_3dgut.py", line 163, in setup_3dgut
    torch.utils.cpp_extension.load(
  File "<DELETED>/.conda/envs/3dgrut/lib/python3.11/site-packages/torch/utils/cpp_extension.py", line 1308, in load
    return _jit_compile(
           ^^^^^^^^^^^^^
  File "<DELETED>/.conda/envs/3dgrut/lib/python3.11/site-packages/torch/utils/cpp_extension.py", line 1710, in _jit_compile
    _write_ninja_file_and_build_library(
  File "<DELETED>/.conda/envs/3dgrut/lib/python3.11/site-packages/torch/utils/cpp_extension.py", line 1823, in _write_ninja_file_and_build_library
    _run_ninja_build(
  File "<DELETED>/.conda/envs/3dgrut/lib/python3.11/site-packages/torch/utils/cpp_extension.py", line 2116, in _run_ninja_build
    raise RuntimeError(message) from e
RuntimeError: Error building extension 'lib3dgut_cc'

Set the environment variable HYDRA_FULL_ERROR=1 for a complete stack trace.
```



# Possible Solution
- Review the compile op of threedgut.slang
  > https://github.com/nv-tlabs/3dgrut/blob/9d2404936e068db60b39fa77da00f8df1b084d6a/threedgut_tracer/setup_3dgut.py#L144-L158
- Alternatively, if the current definition is intentional, update the usage to align with the definition.

--- a/threedgut_tracer/include/3dgut/kernels/cuda/models/gaussianParticles.cuh
+++ b/threedgut_tracer/include/3dgut/kernels/cuda/models/gaussianParticles.cuh
@@ -25,6 +25,13 @@ struct ParticleDensity {
     float padding;
 };
 
+struct ParticeFetchedDensity {
+    float3 position;
+    float3 scale;
+    float33 rotationT;
+    float density;
+};
+
 __forceinline__ __device__ void rotationMatrixTranspose(const float4& q, float33& ret) {
     const float r = q.x;
     const float x = q.y;

--- a/threedgut_tracer/include/3dgut/kernels/cuda/models/shRadiativeGaussianParticles.cuh
+++ b/threedgut_tracer/include/3dgut/kernels/cuda/models/shRadiativeGaussianParticles.cuh
@@ -36,69 +36,70 @@ template <typename TBuffer, bool TDifferentiable>
 struct ShRadiativeGaussianParticlesOptionalBuffer<TBuffer, TDifferentiable, true> : ShRadiativeGaussianParticlesBuffer<TBuffer, TDifferentiable> {
 };
 
-template <typename TDensityParameters,
-          typename TFeaturesParameters,
-          typename Params,
+template <typename Params,
           typename ExtParams,
           bool TDifferentiable = true>
 struct ShRadiativeGaussianVolumetricFeaturesParticles : Params, public ExtParams {
 
-    using DensityParameters    = TDensityParameters;
-    using DensityRawParameters = gaussianParticle_RawParameters_0;
+    using DensityParameters    = threedgut::ParticeFetchedDensity;
+    using DensityRawParameters = threedgut::ParticleDensity;
 
     __forceinline__ __device__ void initializeDensity(threedgut::MemoryHandles parameters) {
+        static_assert(sizeof(DensityRawParameters) == sizeof(gaussianParticle_RawParameters_0), "Sizes must match for binary compatibility");
+        static_assert(sizeof(DensityParameters) == sizeof(gaussianParticle_Parameters_0), "Sizes must match for binary compatibility");
         m_densityRawParameters.ptr =
-            parameters.bufferPtr<gaussianParticle_RawParameters_0>(Params::DensityRawParametersBufferIndex);
+            parameters.bufferPtr<DensityRawParameters>(Params::DensityRawParametersBufferIndex);
     }
 
     __forceinline__ __device__ void initializeDensityGradient(threedgut::MemoryHandles parametersGradient) {
         if constexpr (TDifferentiable) {
             m_densityRawParameters.gradPtr =
-                parametersGradient.bufferPtr<gaussianParticle_RawParameters_0>(Params::DensityRawParametersGradientBufferIndex);
+                parametersGradient.bufferPtr<DensityRawParameters>(Params::DensityRawParametersGradientBufferIndex);
         }
     };
 
     __forceinline__ __device__ DensityRawParameters fetchDensityRawParameters(uint32_t particleIdx) const {
-
-        return *reinterpret_cast<const DensityRawParameters*>(&m_densityRawParameters.ptr[particleIdx]);
+        return m_densityRawParameters.ptr[particleIdx];
     }
 
-    __forceinline__ __device__ TDensityParameters fetchDensityParameters(uint32_t particleIdx) const {
-
-        return particleDensityParameters(particleIdx, {m_densityRawParameters.ptr, nullptr});
+    __forceinline__ __device__ DensityParameters fetchDensityParameters(uint32_t particleIdx) const {
+        const auto parameters = particleDensityParameters(
+            particleIdx,
+            {reinterpret_cast<gaussianParticle_RawParameters_0*>(m_densityRawParameters.ptr), nullptr});
+        return *reinterpret_cast<const DensityParameters*>(&parameters);
     }
 
     __forceinline__ __device__ tcnn::vec3 fetchPosition(uint32_t particleIdx) const {
-        return *(reinterpret_cast<const tcnn::vec3*>(&m_densityRawParameters.ptr[particleIdx].position_0));
+        return *(reinterpret_cast<const tcnn::vec3*>(&m_densityRawParameters.ptr[particleIdx].position));
     }
 
-    __forceinline__ __device__ const tcnn::vec3& position(const TDensityParameters& parameters) const {
-        return *(reinterpret_cast<const tcnn::vec3*>(&parameters.position_1));
+    __forceinline__ __device__ const tcnn::vec3& position(const DensityParameters& parameters) const {
+        return *(reinterpret_cast<const tcnn::vec3*>(&parameters.position));
     }
 
-    __forceinline__ __device__ const tcnn::vec3& scale(const TDensityParameters& parameters) const {
-        return *(reinterpret_cast<const tcnn::vec3*>(&parameters.scale_1));
+    __forceinline__ __device__ const tcnn::vec3& scale(const DensityParameters& parameters) const {
+        return *(reinterpret_cast<const tcnn::vec3*>(&parameters.scale));
     }
 
-    __forceinline__ __device__ const tcnn::mat3& rotation(const TDensityParameters& parameters) const {
+    __forceinline__ __device__ const tcnn::mat3& rotation(const DensityParameters& parameters) const {
         // slang uses row-major order (tcnn uses column-major order), so we return the rotation (not transposed)
-        return *(reinterpret_cast<const tcnn::mat3*>(&parameters.rotationT_0));
+        return *(reinterpret_cast<const tcnn::mat3*>(&parameters.rotationT));
     }
 
-    __forceinline__ __device__ const float& opacity(const TDensityParameters& parameters) const {
-        return parameters.density_1;
+    __forceinline__ __device__ const float& opacity(const DensityParameters& parameters) const {
+        return parameters.density;
     }
 
     __forceinline__ __device__ bool densityHit(const tcnn::vec3& rayOrigin,
                                                const tcnn::vec3& rayDirection,
-                                               const TDensityParameters& parameters,
+                                               const DensityParameters& parameters,
                                                float& alpha,
                                                float& depth,
                                                tcnn::vec3* normal = nullptr) const {
 
         return particleDensityHit(*reinterpret_cast<const float3*>(&rayOrigin),
                                   *reinterpret_cast<const float3*>(&rayDirection),
-                                  parameters,
+                                  reinterpret_cast<const gaussianParticle_Parameters_0&>(parameters),
                                   &alpha,
                                   &depth,
                                   normal != nullptr,
@@ -129,7 +130,7 @@ struct ShRadiativeGaussianVolumetricFeaturesParticles : Params, public ExtParams
         return particleDensityProcessHitFwdFromBuffer(*reinterpret_cast<const float3*>(&rayOrigin),
                                                       *reinterpret_cast<const float3*>(&rayDirection),
                                                       particleIdx,
-                                                      {{m_densityRawParameters.ptr, nullptr, true}},
+                                                      {{reinterpret_cast<gaussianParticle_RawParameters_0*>(m_densityRawParameters.ptr), nullptr, true}},
                                                       &transmittance,
                                                       &integratedDepth,
                                                       integratedNormal != nullptr,
@@ -156,7 +157,9 @@ struct ShRadiativeGaussianVolumetricFeaturesParticles : Params, public ExtParams
             particleDensityProcessHitBwdToBuffer(*reinterpret_cast<const float3*>(&rayOrigin),
                                                  *reinterpret_cast<const float3*>(&rayDirection),
                                                  particleIdx,
-                                                 {{m_densityRawParameters.ptr, m_densityRawParameters.gradPtr, exclusiveGradient}},
+                                                 {{reinterpret_cast<gaussianParticle_RawParameters_0*>(m_densityRawParameters.ptr),
+                                                   reinterpret_cast<gaussianParticle_RawParameters_0*>(m_densityRawParameters.gradPtr),
+                                                   exclusiveGradient}},
                                                  alpha,
                                                  alphaGrad,
                                                  &transmittance,
@@ -181,7 +184,7 @@ struct ShRadiativeGaussianVolumetricFeaturesParticles : Params, public ExtParams
         return particleDensityHitCustom(*reinterpret_cast<const float3*>(&rayOrigin),
                                         *reinterpret_cast<const float3*>(&rayDirection),
                                         particleIdx,
-                                        {{m_densityRawParameters.ptr, nullptr, true}},
+                                        {{reinterpret_cast<gaussianParticle_RawParameters_0*>(m_densityRawParameters.ptr), nullptr, true}},
                                         minHitDistance,
                                         maxHitDistance,
                                         maxParticleSquaredDistance,
@@ -208,7 +211,8 @@ struct ShRadiativeGaussianVolumetricFeaturesParticles : Params, public ExtParams
                                                                    const tcnn::vec3& sourcePosition)
 
     {
-        const auto incidentDirection = particleDensityIncidentDirection(parameters, *reinterpret_cast<const float3*>(&sourcePosition));
+        const auto incidentDirection = particleDensityIncidentDirection(reinterpret_cast<const gaussianParticle_Parameters_0&>(parameters),
+                                                                        *reinterpret_cast<const float3*>(&sourcePosition));
         return *reinterpret_cast<const tcnn::vec3*>(&incidentDirection);
     }
 
@@ -218,11 +222,13 @@ struct ShRadiativeGaussianVolumetricFeaturesParticles : Params, public ExtParams
 
     {
         particleDensityIncidentDirectionBwdToBuffer(particlesIdx,
-                                                    {{m_densityRawParameters.ptr, m_densityRawParameters.gradPtr, exclusiveGradient}},
+                                                    {{reinterpret_cast<gaussianParticle_RawParameters_0*>(m_densityRawParameters.ptr),
+                                                      reinterpret_cast<gaussianParticle_RawParameters_0*>(m_densityRawParameters.gradPtr),
+                                                      exclusiveGradient}},
                                                     *reinterpret_cast<const float3*>(&sourcePosition));
     }
 
-    using FeaturesParameters = TFeaturesParameters;
+    using FeaturesParameters = shRadiativeParticle_Parameters_0;
     using TFeaturesVec       = typename tcnn::vec<ExtParams::FeaturesDim>;
 
     inline __device__ void initializeFeatures(threedgut::MemoryHandles parameters) {
@@ -348,7 +354,7 @@ struct ShRadiativeGaussianVolumetricFeaturesParticles : Params, public ExtParams
             reinterpret_cast<const float3&>(rayOrigin),
             reinterpret_cast<const float3&>(rayDirection),
             particleIdx,
-            reinterpret_cast<const threedgut::ParticleDensity*>(m_densityRawParameters.ptr),
+            m_densityRawParameters.ptr,
             PerRayRadiance ? reinterpret_cast<const float*>(m_featureRawParameters.ptr) : reinterpret_cast<const float*>(particleFeaturesPtr),
             ExtParams::MinParticleKernelDensity,
             ExtParams::AlphaThreshold,
@@ -433,50 +439,50 @@ struct ShRadiativeGaussianVolumetricFeaturesParticles : Params, public ExtParams
             // Perform warp reduction
 #pragma unroll
             for (int mask = 1; mask < warpSize; mask *= 2) {
-                densityRawParameters.position_0.x += __shfl_xor_sync(0xffffffff, densityRawParameters.position_0.x, mask);
-                densityRawParameters.position_0.y += __shfl_xor_sync(0xffffffff, densityRawParameters.position_0.y, mask);
-                densityRawParameters.position_0.z += __shfl_xor_sync(0xffffffff, densityRawParameters.position_0.z, mask);
-                densityRawParameters.density_0 += __shfl_xor_sync(0xffffffff, densityRawParameters.density_0, mask);
-                densityRawParameters.quaternion_0.x += __shfl_xor_sync(0xffffffff, densityRawParameters.quaternion_0.x, mask);
-                densityRawParameters.quaternion_0.y += __shfl_xor_sync(0xffffffff, densityRawParameters.quaternion_0.y, mask);
-                densityRawParameters.quaternion_0.z += __shfl_xor_sync(0xffffffff, densityRawParameters.quaternion_0.z, mask);
-                densityRawParameters.quaternion_0.w += __shfl_xor_sync(0xffffffff, densityRawParameters.quaternion_0.w, mask);
-                densityRawParameters.scale_0.x += __shfl_xor_sync(0xffffffff, densityRawParameters.scale_0.x, mask);
-                densityRawParameters.scale_0.y += __shfl_xor_sync(0xffffffff, densityRawParameters.scale_0.y, mask);
-                densityRawParameters.scale_0.z += __shfl_xor_sync(0xffffffff, densityRawParameters.scale_0.z, mask);
+                densityRawParameters.position.x += __shfl_xor_sync(0xffffffff, densityRawParameters.position.x, mask);
+                densityRawParameters.position.y += __shfl_xor_sync(0xffffffff, densityRawParameters.position.y, mask);
+                densityRawParameters.position.z += __shfl_xor_sync(0xffffffff, densityRawParameters.position.z, mask);
+                densityRawParameters.density += __shfl_xor_sync(0xffffffff, densityRawParameters.density, mask);
+                densityRawParameters.quaternion.x += __shfl_xor_sync(0xffffffff, densityRawParameters.quaternion.x, mask);
+                densityRawParameters.quaternion.y += __shfl_xor_sync(0xffffffff, densityRawParameters.quaternion.y, mask);
+                densityRawParameters.quaternion.z += __shfl_xor_sync(0xffffffff, densityRawParameters.quaternion.z, mask);
+                densityRawParameters.quaternion.w += __shfl_xor_sync(0xffffffff, densityRawParameters.quaternion.w, mask);
+                densityRawParameters.scale.x += __shfl_xor_sync(0xffffffff, densityRawParameters.scale.x, mask);
+                densityRawParameters.scale.y += __shfl_xor_sync(0xffffffff, densityRawParameters.scale.y, mask);
+                densityRawParameters.scale.z += __shfl_xor_sync(0xffffffff, densityRawParameters.scale.z, mask);
             }
 
             // First thread in the warp performs the atomic add
             if ((tileThreadIdx & (warpSize - 1)) == 0) {
-                atomicAdd(&m_densityRawParameters.gradPtr[particleIdx].position_0.x, densityRawParameters.position_0.x);
-                atomicAdd(&m_densityRawParameters.gradPtr[particleIdx].position_0.y, densityRawParameters.position_0.y);
-                atomicAdd(&m_densityRawParameters.gradPtr[particleIdx].position_0.z, densityRawParameters.position_0.z);
-                atomicAdd(&m_densityRawParameters.gradPtr[particleIdx].density_0, densityRawParameters.density_0);
-                atomicAdd(&m_densityRawParameters.gradPtr[particleIdx].quaternion_0.x, densityRawParameters.quaternion_0.x);
-                atomicAdd(&m_densityRawParameters.gradPtr[particleIdx].quaternion_0.y, densityRawParameters.quaternion_0.y);
-                atomicAdd(&m_densityRawParameters.gradPtr[particleIdx].quaternion_0.z, densityRawParameters.quaternion_0.z);
-                atomicAdd(&m_densityRawParameters.gradPtr[particleIdx].quaternion_0.w, densityRawParameters.quaternion_0.w);
-                atomicAdd(&m_densityRawParameters.gradPtr[particleIdx].scale_0.x, densityRawParameters.scale_0.x);
-                atomicAdd(&m_densityRawParameters.gradPtr[particleIdx].scale_0.y, densityRawParameters.scale_0.y);
-                atomicAdd(&m_densityRawParameters.gradPtr[particleIdx].scale_0.z, densityRawParameters.scale_0.z);
+                atomicAdd(&m_densityRawParameters.gradPtr[particleIdx].position.x, densityRawParameters.position.x);
+                atomicAdd(&m_densityRawParameters.gradPtr[particleIdx].position.y, densityRawParameters.position.y);
+                atomicAdd(&m_densityRawParameters.gradPtr[particleIdx].position.z, densityRawParameters.position.z);
+                atomicAdd(&m_densityRawParameters.gradPtr[particleIdx].density, densityRawParameters.density);
+                atomicAdd(&m_densityRawParameters.gradPtr[particleIdx].quaternion.x, densityRawParameters.quaternion.x);
+                atomicAdd(&m_densityRawParameters.gradPtr[particleIdx].quaternion.y, densityRawParameters.quaternion.y);
+                atomicAdd(&m_densityRawParameters.gradPtr[particleIdx].quaternion.z, densityRawParameters.quaternion.z);
+                atomicAdd(&m_densityRawParameters.gradPtr[particleIdx].quaternion.w, densityRawParameters.quaternion.w);
+                atomicAdd(&m_densityRawParameters.gradPtr[particleIdx].scale.x, densityRawParameters.scale.x);
+                atomicAdd(&m_densityRawParameters.gradPtr[particleIdx].scale.y, densityRawParameters.scale.y);
+                atomicAdd(&m_densityRawParameters.gradPtr[particleIdx].scale.z, densityRawParameters.scale.z);
             }
         } else {
-            atomicAdd(&m_densityRawParameters.gradPtr[particleIdx].position_0.x, densityRawParameters.position_0.x);
-            atomicAdd(&m_densityRawParameters.gradPtr[particleIdx].position_0.y, densityRawParameters.position_0.y);
-            atomicAdd(&m_densityRawParameters.gradPtr[particleIdx].position_0.z, densityRawParameters.position_0.z);
-            atomicAdd(&m_densityRawParameters.gradPtr[particleIdx].density_0, densityRawParameters.density_0);
-            atomicAdd(&m_densityRawParameters.gradPtr[particleIdx].quaternion_0.x, densityRawParameters.quaternion_0.x);
-            atomicAdd(&m_densityRawParameters.gradPtr[particleIdx].quaternion_0.y, densityRawParameters.quaternion_0.y);
-            atomicAdd(&m_densityRawParameters.gradPtr[particleIdx].quaternion_0.z, densityRawParameters.quaternion_0.z);
-            atomicAdd(&m_densityRawParameters.gradPtr[particleIdx].quaternion_0.w, densityRawParameters.quaternion_0.w);
-            atomicAdd(&m_densityRawParameters.gradPtr[particleIdx].scale_0.x, densityRawParameters.scale_0.x);
-            atomicAdd(&m_densityRawParameters.gradPtr[particleIdx].scale_0.y, densityRawParameters.scale_0.y);
-            atomicAdd(&m_densityRawParameters.gradPtr[particleIdx].scale_0.z, densityRawParameters.scale_0.z);
+            atomicAdd(&m_densityRawParameters.gradPtr[particleIdx].position.x, densityRawParameters.position.x);
+            atomicAdd(&m_densityRawParameters.gradPtr[particleIdx].position.y, densityRawParameters.position.y);
+            atomicAdd(&m_densityRawParameters.gradPtr[particleIdx].position.z, densityRawParameters.position.z);
+            atomicAdd(&m_densityRawParameters.gradPtr[particleIdx].density, densityRawParameters.density);
+            atomicAdd(&m_densityRawParameters.gradPtr[particleIdx].quaternion.x, densityRawParameters.quaternion.x);
+            atomicAdd(&m_densityRawParameters.gradPtr[particleIdx].quaternion.y, densityRawParameters.quaternion.y);
+            atomicAdd(&m_densityRawParameters.gradPtr[particleIdx].quaternion.z, densityRawParameters.quaternion.z);
+            atomicAdd(&m_densityRawParameters.gradPtr[particleIdx].quaternion.w, densityRawParameters.quaternion.w);
+            atomicAdd(&m_densityRawParameters.gradPtr[particleIdx].scale.x, densityRawParameters.scale.x);
+            atomicAdd(&m_densityRawParameters.gradPtr[particleIdx].scale.y, densityRawParameters.scale.y);
+            atomicAdd(&m_densityRawParameters.gradPtr[particleIdx].scale.z, densityRawParameters.scale.z);
         }
     }
 
 private:
-    ShRadiativeGaussianParticlesBuffer<gaussianParticle_RawParameters_0, TDifferentiable>
+    ShRadiativeGaussianParticlesBuffer<DensityRawParameters, TDifferentiable>
         m_densityRawParameters;
 
     int m_featureActiveShDegree = 0;

--- a/threedgut_tracer/include/3dgut/kernels/cuda/renderers/gutKBufferRenderer.cuh
+++ b/threedgut_tracer/include/3dgut/kernels/cuda/renderers/gutKBufferRenderer.cuh
@@ -339,15 +339,15 @@ struct GUTKBufferRenderer : Params {
 
                 const PrefetchedRawParticleData particleData = prefetchedRawParticlesData[j];
                 if (particleData.idx == GUTParameters::InvalidParticleIdx) {
-                    i = tileNumBlocksToProcess;
+                    ray.kill();
                     break;
                 }
 
                 DensityRawParameters densityRawParametersGrad;
-                densityRawParametersGrad.density_0    = 0.0f;
-                densityRawParametersGrad.position_0   = make_float3(0.0f);
-                densityRawParametersGrad.quaternion_0 = make_float4(0.0f);
-                densityRawParametersGrad.scale_0      = make_float3(0.0f);
+                densityRawParametersGrad.density    = 0.0f;
+                densityRawParametersGrad.position   = make_float3(0.0f);
+                densityRawParametersGrad.quaternion = make_float4(0.0f);
+                densityRawParametersGrad.scale      = make_float3(0.0f);
 
                 TFeaturesVec featuresGrad = TFeaturesVec::zero();
 

--- a/threedgut_tracer/include/3dgut/threedgut.cuh
+++ b/threedgut_tracer/include/3dgut/threedgut.cuh
@@ -39,11 +39,7 @@ struct model_ExternalParams {
 
 #include <3dgut/kernels/cuda/models/shRadiativeGaussianParticles.cuh>
 
-using model_Particles = ShRadiativeGaussianVolumetricFeaturesParticles<gaussianParticle_Parameters_0,
-                                                                       shRadiativeParticle_Parameters_0,
-                                                                       model_InternalParams,
-                                                                       model_ExternalParams,
-                                                                       1>;
+using model_Particles = ShRadiativeGaussianVolumetricFeaturesParticles<model_InternalParams, model_ExternalParams, 1>;
 
 #include <3dgut/kernels/cuda/models/shGaussianModel.cuh>
 

